cli/templateversions.go

```diff
@@ -29,26 +30,68 @@ func (r *RootCmd) templateVersions() *clibase.Cmd {
 		},
 		Children: []*clibase.Cmd{
 			r.templateVersionsList(),
+			r.archiveTemplateVersion(),
+			r.unarchiveTemplateVersion(),
 		},
 	}
 
 	return cmd
 }
 
 func (r *RootCmd) templateVersionsList() *clibase.Cmd {
+	defaultColumns := []string{
+		"Name",
+		"Created At",
+		"Created By",
+		"Status",
+		"Active",
+	}
 	formatter := cliui.NewOutputFormatter(
-		cliui.TableFormat([]templateVersionRow{}, nil),
+		cliui.TableFormat([]templateVersionRow{}, defaultColumns),
 		cliui.JSONFormat(),
 	)
 	client := new(codersdk.Client)
 
+	var includeArchived clibase.Bool
+
 	cmd := &clibase.Cmd{
 		Use: "list <template>",
 		Middleware: clibase.Chain(
 			clibase.RequireNArgs(1),
 			r.InitClient(client),
+			func(next clibase.HandlerFunc) clibase.HandlerFunc {
+				return func(i *clibase.Invocation) error {
+					// This is the only way to dynamically add the "archived"
+					// column if '--include-archived' is true.
+					// It does not make sense to show this column if the
+					// flag is false.
+					if includeArchived {
+						for _, opt := range i.Command.Options {
+							if opt.Flag == "column" {
+								if opt.ValueSource == clibase.ValueSourceDefault {
+									v, ok := opt.Value.(*clibase.StringArray)
+									if ok {
+										// Add the extra new default column.
+										*v = append(*v, "Archived")
+									}
+								}
+								break
+							}
+						}
+					}
+					return next(i)
+				}
+			},
```

Comment:
👀 That's cool, didn't know you could do this.

Reply:
Yea, it's a little janky imo, but the value is tied to the option :cry:.

If we have to do this more often, I can create a helper Middleware to do this more generally.

Reply:
This only adds the column if the user is using the default columns